src/components/views/messages/SenderProfile.tsx

```diff
@@ -100,14 +108,32 @@ export default class SenderProfile extends React.Component {
     render() {
         const {mxEvent} = this.props;
         const colorClass = getUserNameColorClass(mxEvent.getSender());
-        const name = mxEvent.sender ? mxEvent.sender.name : mxEvent.getSender();
         const {msgtype} = mxEvent.getContent();
 
+        const disambiguate = mxEvent.sender?.disambiguate;
+        const displayName = mxEvent.sender?.rawDisplayName || mxEvent.getSender() || "";
+        const mxid = mxEvent.sender?.userId || mxEvent.getSender() || "";
+
         if (msgtype === 'm.emote') {
             return null; // emote message must include the name so don't duplicate it
         }
 
-        let flair = null;
+        const displayNameElement = (
+            <span className={`mx_SenderProfile_displayName ${colorClass}`}>
+                { displayName }
+            </span>
+        );
```

Comment:
can we just inline this into where we render it? Think it'll be cleaner

Reply:
This actually seemed a bit clearer to me this way but sure I'll change it when I get home :D